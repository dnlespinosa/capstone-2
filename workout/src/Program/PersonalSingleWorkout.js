import React, { useEffect, useState } from 'react';


const PersonalSingleWorkout = ({ exercise, setFormData, addData }) => {
    const INTIALSTATE = {
        name: exercise.exercise_name,
        sets:0, 
        reps: 0,
    }

    const [formData2, setFormData2] = useState(INTIALSTATE)
    const [selected, setSelected] = useState(false)
    const handleChange = (e) => {
        const { name, value } = e.target
        setFormData2(formData2 => ({
            ...formData2, 
            [name]: value
        }))
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        console.log(formData2)
        addData(formData2)
        setSelected(true)
        setFormData2(INTIALSTATE)
    }

    return (
        <div>
            {exercise ? 
                <div style={{backgroundColor: 'white', border: '5px solid #2a2a2a', display: 'inline-block', color: '#2a2a2a', padding: '20px', margin: '10px 200px 10px 200px'}}>
                    <div >
                        <h3>{exercise.exercise_name}</h3>
                        <form onSubmit={handleSubmit}>
                            <label htmlFor='sets'>Sets</label>
                            <input 
                                type='number'
                                id='sets'
                                name='sets'
                                onChange={handleChange}></input>
                            <label htmlFor='reps'>Reps</label>
                            <input 
                                type='number'
                                id='reps'
                                name='reps'
                                onChange={handleChange}></input>
                            <button disabled={selected}>Complete Exercise</button>
                        </form>
                        <p>Muscle Group: {exercise.target.Primary}</p>
                        <p>Instructions</p>
                        {exercise.steps.map(step => (
                            <li key={step}>{step}</li>
                        ))}
                    </div>
                    <video controls width="50%" style={{padding: '20px'}}>
                        <source src={exercise.videoURL[0]} type="video/mp4" />
                        Sorry, your browser doesn't support embedded videos.
                    </video>
            </div> : 
            <p style={{color: 'white'}}>Loading Exercise...</p>}
        </div>
        
    )
}

export default PersonalSingleWorkout