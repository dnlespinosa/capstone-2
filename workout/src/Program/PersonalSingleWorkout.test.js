import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import PersonalSingleWorkout from "./PersonalSingleWorkout";
import { UserProvider } from "../testUtils";



describe('<PersonalSingleWorkout />', () => {


    const exercise = {
        Category: 'Barbell', 
        Difficulty: 'Beginner', 
        Force: 'Pull',
        Grips: 'Overhand', 
        exercise_name: 'Barbell Pronated Row', 
        id: 47,
        steps: [
            "Grab the barbell with an overhand grip. Hinge forward while keeping your back flat until your torso is about parallel with the ground.",
            "Let your arms hang freely.",
            "Pull the bar to your torso while tucking your elbows in slightly.",
            "Extend your elbows fully before initiating the next rep."
        ], 
        target: {
            Primary: ['Lats'],
            Secondary: ['Biceps']
        }, 
        videoURL: [
            "https://media.musclewiki.com/media/uploads/videos/branded/male-barbell-pronated-row-front.mp4#t=0.1",
            "https://media.musclewiki.com/media/uploads/videos/branded/male-barbell-pronated-row-side.mp4#t=0.1"
        ]
    }

    // smoke test
    it("renders without crashing", function () {
        render(
            <MemoryRouter>
            <UserProvider>
                <PersonalSingleWorkout exercise={exercise}/>
            </UserProvider>
            </MemoryRouter>,
        );
    });
  
    // snapshot 
    it('matches snapshot', function() {
        const { asFragment } = render(
                <MemoryRouter>
                    <UserProvider>
                        <PersonalSingleWorkout exercise={exercise}/>
                    </UserProvider>
                </MemoryRouter>
            )
        expect(asFragment()).toMatchSnapshot();
        })

    it('adder', () => {
        const { getByText } = render(
            <MemoryRouter>
                <UserProvider>
                    <PersonalSingleWorkout exercise={exercise}/>
                </UserProvider>
            </MemoryRouter>
        )
        console.log(getByText('Complete Exercise'))
    })

})

