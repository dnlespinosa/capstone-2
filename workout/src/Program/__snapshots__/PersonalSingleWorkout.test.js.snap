// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PersonalSingleWorkout /> matches snapshot 1`] = `
<DocumentFragment>
  <div
    id="Barbell Pronated Row"
  >
    <div
      style="background-color: white; border: 5px solid #2a2a2a; display: inline-block; color: rgb(42, 42, 42); padding: 20px; margin: 10px 200px 10px 200px;"
    >
      <div>
        <h3>
          Barbell Pronated Row
        </h3>
        <form>
          <label
            for="sets"
            style="margin: 5px;"
          >
            Sets
          </label>
          <input
            id="sets"
            name="sets"
            style="margin: 5px 20px 5px 0px;"
            type="number"
          />
          <label
            for="reps"
            style="margin: 5px;"
          >
            Reps
          </label>
          <input
            id="reps"
            name="reps"
            style="margin: 5px 20px 5px 0px;"
            type="number"
          />
          <button>
            Complete Exercise
          </button>
        </form>
        <p>
          Muscle Group: Lats
        </p>
        <p>
          Instructions
        </p>
        <div
          style="display: block; margin: 10px; padding: 10px; text-align: center;"
        >
          <li>
            Grab the barbell with an overhand grip. Hinge forward while keeping your back flat until your torso is about parallel with the ground.
          </li>
          <li>
            Let your arms hang freely.
          </li>
          <li>
            Pull the bar to your torso while tucking your elbows in slightly.
          </li>
          <li>
            Extend your elbows fully before initiating the next rep.
          </li>
          <video
            controls=""
            style="padding: 20px;"
            width="50%"
          >
            <source
              src="https://media.musclewiki.com/media/uploads/videos/branded/male-barbell-pronated-row-front.mp4#t=0.1"
              type="video/mp4"
            />
            Sorry, your browser doesn't support embedded videos.
          </video>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
